data/redis.go

```diff
@@ -80,32 +80,107 @@ func (r *RedisConnector) Id() string {
 
 // connectTask is the function that tries to establish a Redis connection (and pings to verify).
 func (r *RedisConnector) connectTask(ctx context.Context) error {
-	options := &redis.Options{
-		Addr:         r.cfg.Addr,
-		Password:     r.cfg.Password,
-		DB:           r.cfg.DB,
-		PoolSize:     r.cfg.ConnPoolSize,
-		DialTimeout:  r.initTimeout,
-		ReadTimeout:  r.getTimeout,
-		WriteTimeout: r.setTimeout,
-	}
-
-	if r.cfg.TLS != nil && r.cfg.TLS.Enabled {
-		tlsConfig, err := common.CreateTLSConfig(r.cfg.TLS)
+	var options *redis.Options
+	var err error
+	redisURI := strings.TrimSpace(r.cfg.URI)
+
+	if redisURI == "" {
+		// Construct from discrete fields (addr, username, password, db)
+		if strings.TrimSpace(r.cfg.Addr) == "" {
+			return fmt.Errorf("missing Redis connection information: either 'uri' or 'addr' must be supplied")
+		}
+
+		var userInfo string
+		if r.cfg.Username != "" || r.cfg.Password != "" {
+			userInfo = r.cfg.Username + ":" + r.cfg.Password + "@"
+		}
```

Comment:
Will this work for current users with:
```yaml
redis:
  addr: some-redis:6379
  password: some-secret
```

Without username? i.e. what was the default username already set when users don't provide to Redis.options? Probably we should use that in such case to not break current users

Reply:
yes, it will. we cover three scenarios:

* if no username is passed, the `default` username will be applied by default
* if no password is passed, there is a case that the password is set as empty ""
* if username is passed but not password, that is wrong and we cover it in the validation